umrr_ros2_driver/src/smartmicro_radar_node.cpp

```diff
@@ -166,6 +171,126 @@ SmartmicroRadarNode::SmartmicroRadarNode(const rclcpp::NodeOptions & node_option
     m_publishers[i] = create_publisher<sensor_msgs::msg::PointCloud2>(
       "umrr/targets_" + std::to_string(i), sensor.history_size);
   }
+
+  // create a ros2 service to change the radar parameters
+  command_srv_ = create_service<umrr_ros2_msgs::srv::SetParam>(
+    "smartmicro_radar_node/set_radar_parameter",
+    std::bind(
+      &SmartmicroRadarNode::change_radar_mode, this, std::placeholders::_1,
+      std::placeholders::_2));
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Radar parameter services are ready.");
+
+  // create a ros2 service to change the IP address
+  ip_addr_srv_ = create_service<umrr_ros2_msgs::srv::SetIp>(
+    "smartmicro_radar_node/configure_ip_address",
+    std::bind(
+      &SmartmicroRadarNode::change_ip_address, this, std::placeholders::_1,
+      std::placeholders::_2));
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Radar is ready to be configured.");
+}
+
+void SmartmicroRadarNode::change_radar_mode(
+  const std::shared_ptr<umrr_ros2_msgs::srv::SetParam::Request> request,
+  std::shared_ptr<umrr_ros2_msgs::srv::SetParam::Response> result)
+{
+  std::shared_ptr<InstructionServiceIface> inst{m_services->GetInstructionService()};
+  std::this_thread::sleep_for(std::chrono::seconds(2));
+  std::shared_ptr<InstructionBatch> batch(inst->AllocateInstructionBatch());
+  std::shared_ptr<SetParamRequest<uint8_t>> radar_mode(
+    new SetParamRequest<uint8_t>(request->section, request->param, request->value)
+  );
+
+  batch->AddRequest(radar_mode);
+  // create list of clients to pass the instructions
+  std::set<ClientId> clients;
+  clients.insert(request->sensor_id);
+  // send instruction batch to the device
+  if (ERROR_CODE_OK != inst->SetInstructionBatch(
+      clients,
+      batch,
+      std::bind(
+        &SmartmicroRadarNode::MyResponseCallback, this, std::placeholders::_1,
+        std::placeholders::_2)))
+  {
+    throw std::runtime_error("Failed to send batch");
```

Comment:
Should this really `throw` (kill the node) if it fails or just output a logged error?